### ISSUE TITLE
Update method doesn't take a simple network structure

### ISSUE THREAD

**Issue by clincha** [clincha-org/proxmox-api#5]
When I pass a full network struct into the update network function I get an error. The error explains that it's not expecting all the fields that I have passed into it. I need to make the update function robust enough to handle updates without the passing call needing to understand which fields need to be passed into it

**Comment by clincha** [comment]
There is a difference between the information the API responds with and the information the API requests.

I think I need to have a couple of different structures, one representing the Network and one representing the Network API Request. That will need a translation layer that takes a network and creates a network request. The response will map onto a network.

It looks like the request for creating and updating the network are the same so this can be a single structure.

**Comment by clincha** [comment]
The response that we get from the network API also needs structures that wrap the network because of the "data" object that is returned as part of the response. This could be renamed to make it clear that there are structures that deal with the communication and structures that deal with the representation of the element. This might become a pattern moving forward and it may make sense to put these in their own files for clarity

**Comment by clincha** [comment]
I have created a structure for network requests which is separate to network. I am now realising that when I want to create a network request I need to map over ~~all the fields from the network that can be accepted~~ the interface name and the type.

**Comment by clincha** [comment]
By creating the `NetworkRequest` struct, I've realised that the `Network` struct does not have all the fields that can be returned by the API. I also don't know an easy way to get that information... 

The documentation tells me all the fields that I can request from the endpoint but not all the fields that come back from the GET endpoint. It does show me the mandatory responses but not the optional ones. For example, I set the comment as part of a test. However, when I do a GET there isn't a field in the struct for it to be mapped to. 

I'm going to get all of the fields that I know of that can be set and then create some logging to warn when something isn't mapped.

**Comment by clincha** [comment]
I have found the `DisallowUnknownFields()` function of the JSON Decoder package. It does what I want it to do but it only tells me a single field at a time. I've tested by setting both the `comments` and `mtu` field and then trying to read it into a struct that doesn't have those fields. It gives me an error like this back:

```text
2024/05/07 21:47:48 WARN json: unknown field "mtu"
```

That's great but then I remove `mtu` from the request and I get a similar error back:

```text
2024/05/07 21:57:21 WARN json: unknown field "comments"
```

Which means having multiple fields in the response that I don't know about will suppress all but one of them. I might be able to write a loop that keeps decoding and removing the field that isn't being mapped until I don't get any fields that don't map. However, do I really want to run that every time a request is made? A better solution would be to find all the possible values the API can return and build my struct to that specification.

**Comment by clincha** [comment]
I have written a nasty bit of code but it works. 


```go

var networkJSON map[string]map[string]interface{}

err = json.Unmarshal(body, &networkJSON)
if err != nil {
	slog.Error(fmt.Sprintf("Unable to unmarshall body for missing field check, error was %v", err))
}

networkModelStruct := reflect.ValueOf(&network).Elem()
var networkModelStructFields []string
for i := 0; i < networkModelStruct.NumField(); i++ {
	networkModelStructFields = append(networkModelStructFields, strings.Replace(networkModelStruct.Type().Field(i).Tag.Get("json"), ",omitempty", "", -1))
}

for name, _ := range networkJSON["data"] {
	if !slices.Contains(networkModelStructFields, name) {
		slog.Warn(fmt.Sprintf("field %q returned by Proxmox API but field does not exist in NetworkModel struct. Please report this to the developers. See: https://github.com/clincha-org/proxmox-api/issues/5", name))
	}
}
```

1.Take the response that we got from the Proxmox API and convert it into JSON
2. Get the names of the fields in the `Network` struct
3. Check if each field returned by Proxmox is inside the `Network` struct
4. If a field isn't in the struct, log a warning to alert me so I can fix the struct

**Comment by clincha** [comment]
Closed by d5d526a893b86d4bccba628244daf0d1f7f628b0
